File: Dagger2Demo/app/src/main/java/org/guildsa/dagger2demo/UserPrefsComponent.java
package org.guildsa.dagger2demo;

import javax.inject.Singleton;
import dagger.Component;

@Singleton @Component(modules = {UserPrefsModule.class})
public interface UserPrefsComponent {

    // We'll call this from the onCreate method of MainActivity to inject
    // UserPrefs into MainActivity.
    void inject(MainActivity activity);
}

File: Dagger2Demo/app/src/main/java/org/guildsa/dagger2demo/UserPrefsComponent.kt
package org.guildsa.dagger2demo

import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [UserPrefsModule::class])
interface UserPrefsComponent {

    // We'll call this from the onCreate method of MainActivity to inject
    // UserPrefs into MainActivity.
    open fun inject(activity: MainActivity?)
}